In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
gdp_df = pd.read_csv("C:/Users/user/Documents/NSS_Projects/un-exploration-ndidiox/data/UNdata_Export_20230915_035202743.csv")

#inspect first 10 rows
gdp_df.head(10)

Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2020  1970.560169              NaN
1     Afghanistan  2019  2065.036235              NaN
2     Afghanistan  2018  2033.804389              NaN
3     Afghanistan  2017  2058.400221              NaN
4     Afghanistan  2016  2057.067978              NaN
5     Afghanistan  2015  2068.265904              NaN
6     Afghanistan  2014  2102.384604              NaN
7     Afghanistan  2013  2116.465258              NaN
8     Afghanistan  2012  2075.491614              NaN
9     Afghanistan  2011  1904.559926              NaN

In [3]:
# inspect last 10 rows
gdp_df.tail(10)

Country or Area  Year        Value  Value Footnotes
7166        Zimbabwe  1999  2872.714795              NaN
7167        Zimbabwe  1998  2915.052992              NaN
7168        Zimbabwe  1997  2856.118188              NaN
7169        Zimbabwe  1996  2808.558556              NaN
7170        Zimbabwe  1995  2573.993785              NaN
7171        Zimbabwe  1994  2603.927369              NaN
7172        Zimbabwe  1993  2420.092791              NaN
7173        Zimbabwe  1992  2437.152908              NaN
7174        Zimbabwe  1991  2733.696626              NaN
7175        Zimbabwe  1990  2652.129209              NaN

In [4]:
# drop Value Footnotes Column
gdp_df = gdp_df.drop('Value Footnotes', axis=1)

In [5]:
gdp_df

Country or Area  Year        Value
0        Afghanistan  2020  1970.560169
1        Afghanistan  2019  2065.036235
2        Afghanistan  2018  2033.804389
3        Afghanistan  2017  2058.400221
4        Afghanistan  2016  2057.067978
...              ...   ...          ...
7171        Zimbabwe  1994  2603.927369
7172        Zimbabwe  1993  2420.092791
7173        Zimbabwe  1992  2437.152908
7174        Zimbabwe  1991  2733.696626
7175        Zimbabwe  1990  2652.129209

[7176 rows x 3 columns]

In [6]:
# rename columns
gdp_df = gdp_df.rename(columns={'Country or Area': 'Country', 'Value': 'GDP_Per_Capita'})

gdp_df

Country  Year  GDP_Per_Capita
0     Afghanistan  2020     1970.560169
1     Afghanistan  2019     2065.036235
2     Afghanistan  2018     2033.804389
3     Afghanistan  2017     2058.400221
4     Afghanistan  2016     2057.067978
...           ...   ...             ...
7171     Zimbabwe  1994     2603.927369
7172     Zimbabwe  1993     2420.092791
7173     Zimbabwe  1992     2437.152908
7174     Zimbabwe  1991     2733.696626
7175     Zimbabwe  1990     2652.129209

[7176 rows x 3 columns]

In [7]:
gdp_df.shape

(7176, 3)

#### The gdp_df dataframe has 7176 rows and 3 columns.

In [8]:
gdp_df.dtypes

Country            object
Year                int64
GDP_Per_Capita    float64
dtype: object

#### The 'Country" column is an object data type, the 'Year' column is an integer data type, and the 'GDP_Per_Capita' column is a float data type. All are the expected data types.

In [14]:
# find years represented in dataset
year = gdp_df["Year"].unique()
year

array([2020, 2019, 2018, 2017, 2016, 2015, 2014, 2013, 2012, 2011, 2010,
       2009, 2008, 2007, 2006, 2005, 2004, 2003, 2002, 2001, 2000, 1999,
       1998, 1997, 1996, 1995, 1994, 1993, 1992, 1991, 1990], dtype=int64)

#### The years 1990 - 2020 are represented in this dataframe. (come back to question 7 )

In [23]:
# 8. number of countries in dataset
countries = gdp_df["Country"].nunique()
countries

242

#### There are 242 countries represented in the dataset. (come back to question 8)

In [29]:
# 9. 
gdp_2014 = gdp_df[gdp_df["Year"] == 2014]
gdp_2014

Country  Year  GDP_Per_Capita
6                     Afghanistan  2014     2102.384604
25    Africa Eastern and Southern  2014     3677.319414
56     Africa Western and Central  2014     4215.646000
87                        Albania  2014    11586.833878
118                       Algeria  2014    11512.690472
...                           ...   ...             ...
7031                      Vietnam  2014     7601.858491
7062           West Bank and Gaza  2014     5967.073437
7089                        World  2014    15224.439424
7120                       Zambia  2014     3450.037600
7151                     Zimbabwe  2014     2357.651166

[242 rows x 3 columns]